source/common/http/http2/codec_impl.cc

```diff
@@ -1136,9 +1153,12 @@ ClientConnectionImpl::ClientConnectionImpl(
                      max_response_headers_count),
       callbacks_(callbacks) {
   ClientHttp2Options client_http2_options(http2_options);
-  nghttp2_session_client_new2(&session_, http2_callbacks_.callbacks(), base(),
-                              client_http2_options.options());
-  sendSettings(http2_options, true);
+  Nghttp2SessionFactory* session_factory =
+      Registry::FactoryRegistry<Nghttp2SessionFactory>::getFactory("envoy.http2.session");
+  ASSERT(session_factory != nullptr);
```

Comment:
nit: this will crash in an obvious way on the next line so I would not bother with this assert.

Reply:
This is now obsolete.